engines/order_management/app/services/order_management/reports/enterprise_fee_summary/scope.rb

```diff
@@ -336,6 +336,10 @@ def filter_by_fee(params)
             if params.payment_method_ids.present?
         end
 
+        def exclude_groups_with_zero_total
+          filter_scope("spree_adjustments.amount != 0")
```

Comment:
I dont think you need to replicate model logic here, adjustments.set_eligibility is already giving you this:
https://github.com/openfoodfoundation/spree/blob/step-6a/core/app/models/spree/adjustment.rb#L65

```suggestion
          filter_scope("spree_adjustments.eligible = true")
```

Reply:
Good catch. Then this can be turned into `filter_scope(eligible: true)` :ok_hand:

Reply:
Thanks, @luisramos0.

My finding about this is that $0 adjustments are not automatically set to `eligible: false` if `mandatory: true`, which is the case for all/most types of order adjustments. The only adjustment I hadn't checked was `TagRule` (I haven't explored how to use this yet). Failed payment transaction fee adjustments are also set to `eligible: false`.

Do let me know if it seems I'm misunderstanding.

In [`CalculatedAdjustments#create_adjustment`](https://github.com/openfoodfoundation/spree/blob/step-6a/core/lib/spree/core/calculated_adjustments.rb#L27):

```ruby
def create_adjustment(label, target, calculable, mandatory=false)
  # ...
end
```

And notice the 4th argument below:

```
$ git grep "create_adjustment("
app/models/product_distribution.rb:20:    a = enterprise_fee.create_adjustment(adjustment_label_for(line_item), line_item.order, line_item, true)
app/models/spree/payment_decorator.rb:24:        payment_method.create_adjustment(adjustment_label, order, self, true)
app/models/tag_rule/discount_order.rb:8:    create_adjustment(I18n.t("discount"), subject, subject)
lib/open_food_network/enterprise_fee_applicator.rb:4:      a = enterprise_fee.create_adjustment(line_item_adjustment_label, line_item.order, line_item, true)
lib/open_food_network/enterprise_fee_applicator.rb:12:      a = enterprise_fee.create_adjustment(order_adjustment_label, order, order, true)
spec/models/enterprise_fee_spec.rb:112:      pd1.enterprise_fee.create_adjustment('foo1', line_item.order, line_item, true)
spec/models/enterprise_fee_spec.rb:113:      pd2.enterprise_fee.create_adjustment('foo2', line_item.order, line_item, true)
spec/models/enterprise_fee_spec.rb:125:      tax_rate.create_adjustment('foo', line_item.order, line_item)
spec/models/enterprise_fee_spec.rb:146:      pd1.enterprise_fee.create_adjustment('foo1', line_item1.order, line_item1, true)
spec/models/enterprise_fee_spec.rb:147:      pd2.enterprise_fee.create_adjustment('foo2', line_item1.order, line_item1, true)
spec/models/enterprise_fee_spec.rb:148:      pd3.enterprise_fee.create_adjustment('foo3', line_item2.order, line_item2, true)
spec/models/enterprise_fee_spec.rb:149:      pd4.enterprise_fee.create_adjustment('foo4', line_item2.order, line_item2, true)
spec/models/product_distribution_spec.rb:77:        adjustment = pd.enterprise_fee.create_adjustment('foo', line_item.order, line_item, true)
spec/models/product_distribution_spec.rb:85:        pd.enterprise_fee.create_adjustment('one', line_item.order, line_item, true)
spec/models/product_distribution_spec.rb:86:        pd.enterprise_fee.create_adjustment('two', line_item.order, line_item, true)
spec/models/spree/order_spec.rb:145:      a = ef.create_adjustment("adjustment", o, o, true)
spec/models/spree/order_spec.rb:153:      a = ef.create_adjustment("adjustment", o, o, true)
spec/models/spree/order_spec.rb:163:      sm.create_adjustment("adjustment", o, o, true)
spec/models/spree/order_spec.rb:171:      ef.create_adjustment("adjustment", li.order, li, true)
spec/models/tag_rule/discount_order_spec.rb:78:        shipping_method.create_adjustment("Shipping", order, order, true)
```

Reply:
Interestingly, Spree's default `mandatory` is `false`.

Reply:
yeah, makes sense.
I am not sure we are using mandatory correctly in these cases...

anyway, we need to keep your original approach here as we don't want to review the mandatory situation just because of this...